File: src/content/docs/topics/cs/coding-problems/stack/1047-remove-all-adjacent-duplicates.go
package main

import "fmt"

func assert(condition bool, msgs ...string) {
	if !condition {
		msg := "assertion failed"
		if len(msgs) > 0 {
			msg = msgs[0]
		}
		panic(msg)
	}
}

func removeDuplicates(s string) string {
	// TODO: implement
	return ""
}

func runTests() {
	assert(removeDuplicates("abbaca") == "ca")
	assert(removeDuplicates("azxxzy") == "ay")
	assert(removeDuplicates("a") == "a")
	assert(removeDuplicates("aa") == "")
	assert(removeDuplicates("abcd") == "abcd")
	fmt.Println("all tests pass")
}

func main() { runTests() }

File: src/content/docs/topics/cs/coding-problems/stack/1047-remove-all-adjacent-duplicates.ts
function assert(condition: boolean, msg: string = ''): void {
    if (!condition) throw new Error(msg || 'Assertion failed');
}

function removeDuplicates(s: string): string {
    // TODO: implement
    return '';
}

function _runTests(): void {
    assert(removeDuplicates('abbaca') === 'ca');
    assert(removeDuplicates('azxxzy') === 'ay');
    assert(removeDuplicates('a') === 'a');
    assert(removeDuplicates('aa') === '');
    assert(removeDuplicates('abcd') === 'abcd');
    console.log('all tests pass');
}

_runTests();
